admin/autoupdater/files/usr/sbin/autoupdater

```diff
@@ -257,21 +259,25 @@ local function autoupdate(mirror)
     return true
   end
 
+  autoupdater_util.run_dir('/usr/lib/autoupdater/download.d')
 
-  os.execute('sync; sysctl -w vm.drop_caches=3')
+  autoupdater_util.exec('sync')
+  autoupdater_util.exec('sysctl', '-w', 'vm.drop_caches=3')
   collectgarbage()
 
   local image = os.tmpname()
   if not fetch_firmware(mirror, manifest.filename, image) then
+    autoupdater_util.run_dir('/usr/lib/autoupdater/abort.d')
     return false
   end
 
-  local popen = io.popen(string.format("sha512sum '%s'", image))
+  local popen = io.popen(string.format("exec sha512sum '%s'", image))
```

Comment:
True, reimplementing io.popen would be the best solution, but seemed like unnecessary work (as the Lua autoupdater will be replaced by C anyways after 2016.1).

The autoupdater_util.exec function, on the other hand, was important, as it is now also used to call the download.d and abort.d executables, which are supplied by other packages and thus might not follow sensible naming conventions ;)